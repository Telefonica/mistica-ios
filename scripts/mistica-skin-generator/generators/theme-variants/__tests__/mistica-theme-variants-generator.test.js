import fs from 'fs';
import {resolve} from 'path';
import {generateMisticaThemeVariants} from '../mistica-theme-variants-generator.js';

test('generateMisticaThemeVariants', () => {
    const anyBrandTokensPath = resolve(__dirname, 'any-brand-tokens.json');
    const anyBrandTokens = JSON.parse(fs.readFileSync(anyBrandTokensPath, 'utf-8'));
    const result = generateMisticaThemeVariants(anyBrandTokens);

    expect(result).toMatchSnapshot();
});
