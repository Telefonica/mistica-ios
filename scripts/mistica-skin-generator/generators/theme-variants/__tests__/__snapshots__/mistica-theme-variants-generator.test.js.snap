// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateMisticaThemeVariants 1`] = `
"
// Generated using Make
// DO NOT EDIT

import Foundation

public protocol MisticaThemeVariants: Sendable {
    var successFeedback: MisticaThemeVariantType { get }
    var brandLoadingScreen: MisticaThemeVariantType { get }
}
"
`;
